src/picosvg/svg_types.py

```diff
@@ -251,6 +306,8 @@ def __iter__(self):
     def walk(self, callback):
         """Walk path and call callback to build potentially new commands.
 
+        Walk may tweak the path even if callback is a nop.
```

Comment:
I don't know if that's a good idea, that walk also tweaks the paths outside the provided callbacks.. It's supposed to just "walk" with a callback, with the latter doing the actual modifications if any. In my other patch I opted for providing all the data for a callback to perform the snapping of the last point, but did not do it within the walk itself.

Reply:
In retrospect agreed, will make walk just walk.